### PR TITLE
DOC:Update the description of set_printoptions in quickstart doc #13900

### LINKED ISSUE #13900: Quickstart guide recommends np.set_printoptions(threshold=np.nan), which fails
We should update [this guide](https://github.com/eric-wieser/numpy/blob/master/doc/source/user/quickstart.rst#printing-arrays) to use `sys.maxsize`.

@mattip, I'm correct in assuming you prefer that to np.inf?

### PR DESCRIPTION
**Pull request by sameshl**
Quickstart guide recommends np.set_printoptions(threshold=np.nan) but that fails.
So changed that to np.set_printoptions(threshold=sys.maxsize)

Closes #13900

### CODE REVIEW

File: numpy/core/arrayprint.py [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+                           flexible)
+from .numeric import concatenate, asarray, errstate
+from .fromnumeric import ravel, any
+from .multiarray import (array, dragon4_positional, dragon4_scientific,
+                         datetime_as_string, datetime_data, ndarray,
+                         set_legacy_print_mode)
+from . import multiarray
+from .umath import absolute, not_equal, isnan, isinf, isfinite, isnat
+from . import numerictypes as _nt
+import numpy as np
```
Comment (mattip) [inline numpy/core/arrayprint.py]: Move these lines back to where they were.

Comment (sameshl) [reply numpy/core/arrayprint.py]: Done. Looks like my text editor Atom made these changes when I saved and I didn't check.

File: numpy/core/arrayprint.py [hunk 2]
```diff
@@ -114,6 +114,7 @@ def set_printoptions(precision=None, threshold=None, edgeitems=None,
     threshold : int, optional
         Total number of array elements which trigger summarization
         rather than full repr (default 1000).
+	Preferred max value to be used is sys.maxsize.
```
Comment (eric-wieser) [inline numpy/core/arrayprint.py]: This phrasing isn't the best. Perhaps ``To always use the full repr without summarization, pass `sys.maxsize`.``

Comment (sameshl) [reply numpy/core/arrayprint.py]: Done. Thank you for the suggestion!

File: numpy/core/arrayprint.py [hunk 3]
```diff
@@ -114,6 +114,7 @@ def set_printoptions(precision=None, threshold=None, edgeitems=None,
     threshold : int, optional
         Total number of array elements which trigger summarization
         rather than full repr (default 1000).
+	To always use the full repr without summarization, pass `sys.maxsize`.
```
Comment (eric-wieser) [inline numpy/core/arrayprint.py]: Indent is wrong here

Comment (sameshl) [reply numpy/core/arrayprint.py]: Fixed.

**Review by eric-wieser — APPROVED** [review]

**Comment by seberg** [comment]
Thanks @sameshl and Eric.

**Pull request merged**